packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewClientFlutterApiImpl.java

```diff
@@ -26,40 +27,39 @@ public class WebViewClientFlutterApiImpl extends WebViewClientFlutterApi {
   @RequiresApi(api = Build.VERSION_CODES.M)
   static GeneratedAndroidWebView.WebResourceErrorData createWebResourceErrorData(
       WebResourceError error) {
-    final GeneratedAndroidWebView.WebResourceErrorData errorData =
-        new GeneratedAndroidWebView.WebResourceErrorData();
-    errorData.setErrorCode((long) error.getErrorCode());
-    errorData.setDescription(error.getDescription().toString());
-
-    return errorData;
+    return new GeneratedAndroidWebView.WebResourceErrorData.Builder()
+        .setErrorCode((long) error.getErrorCode())
+        .setDescription(error.getDescription().toString())
+        .build();
   }
 
   @SuppressLint("RequiresFeature")
   static GeneratedAndroidWebView.WebResourceErrorData createWebResourceErrorData(
       WebResourceErrorCompat error) {
-    final GeneratedAndroidWebView.WebResourceErrorData errorData =
-        new GeneratedAndroidWebView.WebResourceErrorData();
-    errorData.setErrorCode((long) error.getErrorCode());
-    errorData.setDescription(error.getDescription().toString());
-
-    return errorData;
+    return new GeneratedAndroidWebView.WebResourceErrorData.Builder()
+        .setErrorCode((long) error.getErrorCode())
+        .setDescription(error.getDescription().toString())
+        .build();
   }
 
   @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
   static GeneratedAndroidWebView.WebResourceRequestData createWebResourceRequestData(
       WebResourceRequest request) {
-    final GeneratedAndroidWebView.WebResourceRequestData requestData =
-        new GeneratedAndroidWebView.WebResourceRequestData();
-    requestData.setUrl(request.getUrl().toString());
-    requestData.setIsForMainFrame(request.isForMainFrame());
+    final GeneratedAndroidWebView.WebResourceRequestData.Builder requestData =
+        new GeneratedAndroidWebView.WebResourceRequestData.Builder()
+            .setUrl(request.getUrl().toString())
+            .setIsForMainFrame(request.isForMainFrame())
+            .setHasGesture(request.hasGesture())
+            .setMethod(request.getMethod())
+            .setRequestHeaders(
+                request.getRequestHeaders() != null
+                    ? request.getRequestHeaders()
+                    : new HashMap<>());
```

Comment:
Apparently `request.getRequestHeaders()` can return null

```
04-27 07:50:01.128: D/AndroidRuntime(21722): Shutting down VM
04-27 07:50:01.128: D/AndroidRuntime(21722): --------- beginning of crash
04-27 07:50:01.128: E/AndroidRuntime(21722): FATAL EXCEPTION: main
04-27 07:50:01.128: E/AndroidRuntime(21722): Process: io.flutter.plugins.webviewflutterandroidexample, PID: 21722
04-27 07:50:01.128: E/AndroidRuntime(21722): java.lang.IllegalStateException: Nonnull field "requestHeaders" is null.
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at io.flutter.plugins.webviewflutter.GeneratedAndroidWebView$WebResourceRequestData.setRequestHeaders(GeneratedAndroidWebView.java:103)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at io.flutter.plugins.webviewflutter.GeneratedAndroidWebView$WebResourceRequestData$Builder.build(GeneratedAndroidWebView.java:161)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at io.flutter.plugins.webviewflutter.WebViewClientFlutterApiImpl.createWebResourceRequestData(WebViewClientFlutterApiImpl.java:58)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at io.flutter.plugins.webviewflutter.WebViewClientFlutterApiImpl.requestLoading(WebViewClientFlutterApiImpl.java:164)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at io.flutter.plugins.webviewflutter.WebViewClientHostApiImpl$WebViewClientImpl.shouldOverrideUrlLoading(WebViewClientHostApiImpl.java:88)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at org.chromium.android_webview.AwContentsClientBridge.shouldOverrideUrlLoading(chromium-TrichromeWebViewGoogle.apk-stable-410410683:16)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at android.os.MessageQueue.nativePollOnce(Native Method)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at android.os.MessageQueue.next(MessageQueue.java:335)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at android.os.Looper.loop(Looper.java:183)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at android.app.ActivityThread.main(ActivityThread.java:7664)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at java.lang.reflect.Method.invoke(Native Method)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
04-27 07:50:01.128: E/AndroidRuntime(21722): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

Reply:
Yes, I remember the old implementation had a PR at some point to fix a null crash with request headers.